add profile support to csv location description (#1349)
cap-js/docs: guides/databases.md

### Location of CSV Files

> CSV files can be located in the folders _db/data_ and _test/data_ as well as in any _data_ folder next to your CDS model files. `cds watch` or `cds deploy` by default load CSV files from _test/data_ while `cds build`, which is used to prepare production deployments, will not. 

Reviewer: ```suggestion
CSV files can be found in the folders _db/data_ and _test/data_, as well as in any _data_ folder next to your CDS model files. When you use `cds watch` or `cds deploy`, CSV files are loaded by default from _test/data_. However, when preparing for production deployments using `cds build`, CSV files from _test/data_ are not loaded.
```

::: details Adding initial data next to your data model
The content of these 'co-located' `.cds` files actually doesn't matter, but they need to be included in your data model, through a `using` clause in another file for example.

> If you want to use some CSV files only for your production deployments, but not in tests, you can do so by adding them to a separate data folder, e.g. _db/hana/data_. Define a _index.cds_ file in the _hana_ folder as described above and configure this model location in a dummy cds service (e.g. _hanaDataSrv_) using the `[production]` profile:

Reviewer: ```suggestion
If you need to use certain CSV files exclusively for your production deployments, but not for tests, you can achieve this by including them in a separate data folder, for example, _db/hana/data_. Create an _index.cds_ file in the _hana_ folder as outlined earlier. Then, set up this model location in a dummy cds service, for example _hanaDataSrv_, using the `[production]` profile.
```

```json
"cds": {
  "requires": {
    "[production]": {
      "hanaDataSrv ": { "model": "hana" }
     }
  }
}
````

> As a consequence `cds build –production` will add the model folder _hana_ while `cds deploy` or `cds watch` will not as the development profile is used by default. You can verify by checking the cds build logs for the hana build task. Of course, this mechanism can also be used for PostgreSQL database deployments.

Reviewer: ```suggestion
As a consequence, when you run `cds build –production` the model folder _hana_ is added, but it's not added when you run `cds deploy` or `cds watch` because the development profile is used by default. You can verify this by checking the cds build logs for the hana build task. Of course, this mechanism can also be used for PostgreSQL database deployments.
```

:::

::: details On SAP HANA ...

> CSV and _hdbtabledata_ files located in the _src_ folder of your database module will be treated as native SAP HANA artifacts and deployed as they are. The advantage of this solution is that you can customize the _hdbtabledata_ files if required (e.g. adding a custom `include_filter` setting if you need to mix initial and customer data in one table). The disadvantage is that you have to redundantly maintain them in order to keep them in-sync with your CSV files.

Reviewer: ```suggestion
CSV and _hdbtabledata_ files found in the _src_ folder of your database module are treated as native SAP HANA artifacts and deployed as they are. This approach offers the advantage of customizing the _hdbtabledata_ files if needed, such as adding a custom `include_filter` setting to mix initial and customer data in one table. However, the downside is that you must redundantly maintain them to keep them in sync with your CSV files.
```
